File: implementations/java-springboot/harness/test/testdata/no-cross-aggregate-reference/good/payment/domain/Payment.java
package com.example.accountservice.payment.domain;

public class Payment {
    private String paymentId;
    private String cardId;
    private String accountId;
    private PaymentStatus status;

    private Payment() {
    }

    public static Payment create(String cardId, String accountId, long amount) {
        return new Payment();
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/no-cross-aggregate-reference/bad-payment-holds-refund/payment/domain/Payment.kt
package com.example.accountservice.payment.domain

class Payment private constructor() {
    var paymentId: String = ""
        private set

    // Violation — directly holds another Aggregate(Refund) as a field. Only ID references like paymentId are allowed.
    var refund: Refund? = null
        private set

    companion object {
        fun create(): Payment = Payment().apply { this.paymentId = "pay-1" }
    }
}
